### PR TITLE
Add Crane Glacier mesh generation

### PR DESCRIPTION
**Pull request by trhille**
This merge adds a Crane Glacier test group with a mesh generation test case.

Checklist
* [x] User's Guide has been updated
* [x] Developer's Guide has been updated
* [x] API documentation in the Developer's Guide (`api.rst`) has any new or modified class, method and/or functions listed
* [x] Documentation has been [built locally](https://mpas-dev.github.io/compass/latest/developers_guide/building_docs.html) and changes look as expected
* [x] Document (in a comment titled `Testing` in this PR) any testing that was used to verify the changes

### CODE REVIEW

**Comment by trhille** [comment]
Testing
-------
I generated a 500m–1km mesh on a Perlmutter login node in 3:26.
image
Note: BedMachine v2 (used here) and v3 both have a large hump in the bed topography, which is almost certainly an artifact. There is a fix in v3.5, which is not yet publicly available.

File: docs/users_guide/landice/test_groups/crane.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    # Bounds of Crane regional mesh
+    x_min = -2.442e6
+    x_max = -2.246e6
+    y_min = 1.124e6
+    y_max = 1.334e6
+    
+    # distance from ice margin to cull (km).
+    # Set to a value <= 0 if you do not want
+    # to cull based on distance from margin.
+    cull_distance = 10.0
```
Comment (matthewhoffman) [inline docs/users_guide/landice/test_groups/crane.rst:42]: minor detail - this differs from the value in the cfg file

**Review by matthewhoffman — APPROVED** [review]
@trhille , thanks for adding this and making such a clean PR.  I noticed one tiny thing that you may want to update.  Let me know when you want me to merge.

**Comment by trhille** [comment]
@matthewhoffman, this is ready, but needs the fixes in https://github.com/MPAS-Dev/MPAS-Tools/pull/551 to pass.

**Pull request merged**